### PR TITLE
Catch empty queries.

### LINKED ISSUE #105: MultiBlast should catch empty sequences.

### PR DESCRIPTION
**Pull request by Foxcapades**
Multi-blast currently allows empty sequences through as long as full query is not empty (the query contains only header lines).

This should catch that issue and prevent empty jobs from making it through.

Resolves #105

### CODE REVIEW

File: rest-service/src/main/kotlin/mb/api/service/valid/SequenceValidator.kt [hunk 1]
```diff
@@ -32,8 +32,11 @@ interface SequenceValidator {
    * @return Whether all characters in the input {@code CharSequence} are valid.
    */
   fun validate(sequence: Int, seq: String): SequenceValidationError? {
+    // Input scanner
     val scan = Scanner(seq)
+    // Current line number
     var lineNum = 1
+    // Number of sequence lines in the input.
```
Comment (ryanrdoherty) [inline rest-service/src/main/kotlin/mb/api/service/valid/SequenceValidator.kt:39]: Are you checking for empty lines below?  Maybe covered by isValid(), but empty IS valid, just should be skipped.  Files can look like this:
```
> header line
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa

> next sequence
fffffffffffffffffffffffffffffffffffffffffffffffff
```
This code might still do what it's supposed to do (catch 0 sequence files) but maybe make explicit if you're going to have a loop and not short-circuit at the first valid sequence line?

**Review by ryanrdoherty — APPROVED** [review]

File: rest-service/src/main/kotlin/mb/api/service/valid/SequenceValidator.kt (hunk 1 above)
Comment (Foxcapades) [reply rest-service/src/main/kotlin/mb/api/service/valid/SequenceValidator.kt:39]: It handles that.  It's counting all sequences in the file, so it will still pass a file like

```
> header 1
> header 2
catcatcat
```

But i considered that to be fine since they will still get a combined result...  However we may want to catch that to prevent issues with the individual result?

Since this code is doing more than just testing for empty sequences we don't short-circuit because we still need to validate the non-empty lines.

Comment (Foxcapades) [reply rest-service/src/main/kotlin/mb/api/service/valid/SequenceValidator.kt:39]: In fact, I'm going to go ahead and reject this PR.  I'll open a new one when i fix that.

**Comment by Foxcapades** [comment]
Reopening as the "missed" case is actually handled.  The query is broken into separate sequences upstream and the `validate` method is only validating a single sequence in the query.

File: rest-service/src/main/kotlin/mb/api/service/valid/SequenceValidator.kt (hunk 1 above)
Comment (Foxcapades) [reply rest-service/src/main/kotlin/mb/api/service/valid/SequenceValidator.kt:39]: On further investigation, it appears this case is actually handled.

**Review by steve-fischer-200 — APPROVED** [review]
